sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -5666,14 +5666,15 @@ atomic_update(AccessorTy acc, Toffset byte_offset, simd_mask<N> mask) {
 /// @return A vector of the old values at the memory locations before the
 ///   update.
 ///
+
 template <atomic_op Op, typename T, int N, typename Toffset,
           typename AccessorTy,
           typename PropertyListT =
               ext::oneapi::experimental::detail::empty_properties_t>
 __ESIMD_API std::enable_if_t<
     __ESIMD_DNS::get_num_args<Op>() == 1 && std::is_integral_v<Toffset> &&
-        sycl::detail::acc_properties::is_accessor_v<AccessorTy> &&
-        !sycl::detail::acc_properties::is_local_accessor_v<AccessorTy> &&
+        detail::is_device_accessor_with_v<
+            AccessorTy, detail::accessor_mode_cap::can_write> &&
```

Comment:
atomic_update requires accessor not only with `write` permissions, but with `read` permissions too.